EIPS/eip-4762.md

```diff
@@ -39,47 +39,43 @@ We define access events as follows. When an access event takes place, the access
 
 #### Access events for account headers
 
-When a non-precompile address is the target of a `CALL`, `CALLCODE`, `DELEGATECALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, or is the target address of a contract creation whose initcode starts execution, process these access events:
+When:
 
-```
-(address, 0, VERSION_LEAF_KEY)
-(address, 0, CODE_SIZE_LEAF_KEY)
-```
+ 1. a non-precompile address is the target of a `CALL`, `CALLCODE`, `DELEGATECALL`, `STATICALL`, `AUTHCALL`, `SELFDESTRUCT`, `EXTCODESIZE`, or `EXTCODECOPY` opcode, 
+ 2. a non-precompile address is the target address of a contract creation whose initcode starts execution,
+ 3. any address is the target of the `BALANCE` opcode
+ 4. a _deployed_ contract calls `CODECOPY`
 
-If a call is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these two access events:
+process this access events:
 
 ```
-(caller_address, 0, BALANCE_LEAF_KEY)
-(callee_address, 0, BALANCE_LEAF_KEY)
+(address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-When a contract is created, process these access events:
+If a `*CALL` or `SELFDESTRUCT` is value-bearing (ie. it transfers nonzero wei), whether or not the callee is a precompile, process these additional two access events:
 
 ```
-(contract_address, 0, VERSION_LEAF_KEY)
-(contract_address, 0, NONCE_LEAF_KEY)
-(contract_address, 0, BALANCE_LEAF_KEY)
-(contract_address, 0, CODE_KECCAK_LEAF_KEY)
-(contract_address, 0, CODE_SIZE_LEAF_KEY)
+(caller_address, 0, BASIC_DATA_LEAF_KEY)
+(callee_address, 0, BASIC_DATA_LEAF_KEY)
 ```
 
-If the `BALANCE` opcode is called targeting some address, process this access event:
-
+When calling `EXTCODEHASH`, process the access event:
 ```
-(address, 0, BALANCE_LEAF_KEY)
+(address, 0, CODE_HASH_LEAF_KEY)
```

Comment:
In 4762 we have mixed namings with `CODE_HASH_LEAF_KEY` and `CODEHASH_LEAF_KEY`.
And in 6800 we use `CODE_KECCAK_LEAF_KEY`.

I'd say to change everything to `CODE_HASH_LEAF_KEY` or `CODEHASH_LEAF_KEY`.

Reply:
I think the big deciding factor is: when ethereum completes its shift to sha256 as its primary hash function (or even moves to something more stark-friendly), would this still be the keccak, or would it change (even if for new contracts or EOF only)? If still keccak, then call it CODE_KECCAK, if not, call it CODE_HASH.

Reply:
using `CODEHASH` for future-proofiness